src/main/java/org/pesmypetcare/webservice/dao/MedicationDaoImpl.java

```diff
@@ -0,0 +1,172 @@
+        db = FirebaseFactory.getInstance().getFirestore();
+
+        DELFAIL_KEY = "deletion-failed";
+        MEDICATION_DOES_NOT_EXIST_EXC = "The med does not exist";
+        INVALID_MEDICATION_EXC = "invalid-med";
+    }
+
+    @Override
+    public void createMedication(String owner, String petName, String dateName, MedicationEntity medication) {
+        DocumentReference medicationRef = getMedicationsRef(owner, petName).document(dateName);
+        medicationRef.set(medication);
+    }
+
+    @Override
+    public void deleteByDateAndName(String owner, String petName, String dateName) {
+        DocumentReference medRef = getMedicationsRef(owner, petName).document(dateName);
+        medRef.delete();
+    }
+
+    @Override
+    public void deleteAllMedications(String owner, String petName) throws DatabaseAccessException {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        try {
+            ApiFuture<QuerySnapshot> future = medicationsRef.get();
+            List<QueryDocumentSnapshot> medicationDocuments = future.get().getDocuments();
+            for (QueryDocumentSnapshot medicationDocument : medicationDocuments) {
+                medicationDocument.getReference().delete();
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+    }
+
+    @Override
+    public MedicationEntity getMedicationData(String owner, String petName, String dateName)
+            throws DatabaseAccessException {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        DocumentReference medicationsRefDoc = medicationsRef.document(dateName);
+        ApiFuture<DocumentSnapshot> future = medicationsRefDoc.get();
+        DocumentSnapshot medicationDoc;
+        try {
+            medicationDoc = future.get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        if (!medicationDoc.exists()) {
+            throw new DatabaseAccessException(INVALID_MEDICATION_EXC, MEDICATION_DOES_NOT_EXIST_EXC);
+        }
+        return medicationDoc.toObject(MedicationEntity.class);
+    }
+
+    @Override
+    public List<Map<List<String>, Object>> getAllMedicationData(String owner, String petName)
+            throws DatabaseAccessException, ExecutionException, InterruptedException {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        List<Map<List<String>, Object>> externalList = new ArrayList<>();
+        String aux;
+            ApiFuture<QuerySnapshot> future = medicationsRef.get();
+            List<QueryDocumentSnapshot> medicationDocuments = future.get().getDocuments();
+            for (QueryDocumentSnapshot medicationDocument : medicationDocuments) {
+                Map<List<String>, Object> internalList = new HashMap<>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['for' child has incorrect indentation level 16, expected level should be 12.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)